Add PHP 7.1 & 7.2 support, drop PHP 5.5 & HHVM, add PHPUnit 6 support

Slamdunk opened the pull request:
- [x] Add PHP 7.1 support ([session.* ini settings](https://secure.php.net/manual/en/migration71.other-changes.php#migration71.other-changes.session-id-generation-without-hashing))
    1. Deprecate session.entropy_file
    1. Deprecate session.entropy_length
    1. Deprecate session.hash_function
    1. Deprecate session.hash_bits_per_character
    1. Support session.sid_length
    1. Support session.sid_bits_per_character 
- [x] Add partial PHP 7.2 support
    1. Fix valid error `Fatal error: Declaration of Zend\Session\AbstractContainer::offsetGet($key) must be compatible with & Zend\Stdlib\ArrayObject::offsetGet($key) in ./zend-session/src/AbstractContainer.php on line 27`
    1. Don't fix any other error until GA release
- [x] Drop HHVM support
- [x] Drop PHP 5.5 support
- [x] Add PHPUnit 6 support

weierophinney committed:
Bumped version

Koopzington committed:
Remove fast_finish flag

It has been confirmed that the fast_finish flag is responsible for triggering multiple build notifications on #zftalk.dev.

MatthiasKuehneEllerhold committed:
Fix the configuration key in the documentation from 'session' to 'session_manager'

Koopzington committed:
Remove CS dependencies and add zend-coding-standard to require-dev

Koopzington committed:
Delete .php_cs

Koopzington committed:
add/update phpcs.xml

Koopzington committed:
add/update composer scripts

Koopzington committed:
Use composer scripts in .travis.yml

Koopzington committed:
Replace mentions of php-cs-fixer with phpcs

Koopzington committed:
Replace mentions of phpcs and phpcbf with composer scripts

rarog committed:
Destroy should only return false when removal of session truely fails.
Finally fixes #36

Koopzington committed:
Automated CS fixes

Koopzington committed:
CS fixes

rarog committed:
Tests for: Destroy should only return false when removal of session truely fails.
Finally fixes #36

rarog committed:
Correcting the used classes.

rarog committed:
Defining behaviour of the test.

rarog committed:
Mocking additional argument.

rarog committed:
Trying callback function.

rarog committed:
Last try for today.

geerteltink committed:
Enable Travis CI slack notifications

Commit:
SessionManager creates instance of $requestedName

Add support for Zend v3 FactoryInterface by considering the
$requestedName parameter when creating a SessionManager instance.

This allows applications to extend SessionManager by still using
the default factory with all its logic.

Slamdunk committed:
Travis: test against 7.1 and nightly

Slamdunk committed:
Distinct PHP 5.6+7.0 from PHP 7.1 exclusive session.* functionality

Slamdunk committed:
Added missing optionsProvider data set

Slamdunk committed:
Travis already do a composer self-update

Slamdunk committed:
Fix PHP 7.2 fatal error

Slamdunk committed:
Travis: remove duplicated envs

Slamdunk committed:
Drop HHVM support

Slamdunk committed:
Drop PHP 5.5 support

Slamdunk committed:
Support PHPUnit 6

Slamdunk committed:
Travis install: can't ignore platform requirements

Slamdunk committed:
SessionManagerTest: use lambda function to test validator valid callable

Slamdunk committed:
Fixed PHPUnit compatibility from 5.7 to 6.1

Slamdunk commented:
Hi, I would appreciate some feedback on this PR.

The code-coverage will always result in a decrease because different code (was added and) is tested in different PHP versions.

samsonasik commented on src/Config/SessionConfig.php:
```diff
@@ -246,6 +263,27 @@ public function setHashBitsPerCharacter($hashBitsPerCharacter)
+    public function setSidBitsPerCharacter($sidBitsPerCharacter)
+    {
+        if (!is_numeric($sidBitsPerCharacter)
+            || !in_array($sidBitsPerCharacter, $this->validSidBitsPerCharacters)
```
`||` should be `&&` ? also, I think in_array can use strict compare with pass `true` in 3rd parameter:  

```php
!in_array($sidBitsPerCharacter, $this->validSidBitsPerCharacters, true)
```

Slamdunk replied:
`||` is correct:
```php
if ($cond1 && $cond2) {
    ok_code();
}
error();
// Otherwise
if (! $cond1 || ! $cond2) {
    error();
}
ok_code();
```
In merit of `in_array` strict parameter, althought I would also advocate the adoption of strictness, for ex. with the [PHP-CS-Fixer\StrictParamFixer](https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.3.2/src/Fixer/Strict/StrictParamFixer.php), we can keep it like this to stay even with [setHashBitsPerCharacter](https://github.com/zendframework/zend-session/blob/release-2.7.3/src/Config/SessionConfig.php#L234-L246) method (that I just copy-pasted) and because `is_numeric` already ensure the check.

Otherwise we can go with strict param and remove `is_numeric` call and optain the same result; in this case I would also go with chainging the all `in_array` occurrencies to keep a standard.

Let me know what maintainers prefer.

weierophinney committed:
Fix signature issue with AbstractContainer::offsetGet

`AbstractContainer` inherits from `Zend\Stdlib\ArrayObject`.
`ArrayObject::offsetGet()` is defined such as to _return by reference_;
however, the override in `AbstractContainer` did not. This casuses an
issue in current alpha versions of PHP 7.2, as it is considered an
incompatible signature.

This patch updates the `AbstractContainer::offsetGet()` declaration to
return by reference. All tests related to containers continue to pass.

weierophinney committed:
Fix failing SessionConfig tests on PHP 7.1+

A number of `session.*` INI options were removed starting in 7.1,
causing tests to fail. This patch updates the test suite to skip
tests for INI options that do not exist when run under 7.1.

weierophinney committed:
Test against PHP 7.1 on Travis

weierophinney committed:
Merge pull request #56 from Koopzington/remove-fast-finish

Remove fast_finish flag

weierophinney committed:
Merge pull request #65 from Koopzington/new-coding-standard

New coding standard

weierophinney committed:
Merge pull request #72 from xtreamwayz/hotfix/slack-notifications

Enable Travis CI slack notifications

Conflicts:
	.travis.yml

weierophinney committed:
Merge branch 'hotfix/qa-tools'

Close #56
Close #65
Close #72

weierophinney committed:
Merge branch 'hotfix/qa-tools' into develop

Forward port #56
Forward port #65
Forward port #72

weierophinney committed:
Merge pull request #61 from MatthiasKuehneEllerhold/docs_fix_config_key

Fix the configuration key in the documentation

weierophinney committed:
Merge branch 'hotfix/61'

Close #61

weierophinney committed:
Merge branch 'hotfix/61' into develop

Forward port #61

weierophinney committed:
Merge pull request #66 from rarog/master

Destroy should only return false when removal of session truely fails.

weierophinney committed:
Updated test prophecy to return by reference

`Cache::getItem()` expects to return the second argument by reference.
Updated the expectation to do this.

weierophinney committed:
Added CHANGELOG for #66

weierophinney committed:
Merge branch 'hotfix/66'

Close #66
Fixes #36

weierophinney committed:
Merge branch 'hotfix/66' into develop

Forward port #66

weierophinney committed:
Added CHANGELOG for #79

weierophinney committed:
Merge branch 'hotfix/signatures' into develop

Forward port #79

Conflicts:
	CHANGELOG.md

weierophinney committed:
Merge pull request #73 from thomascube/master

SessionManager creates instance of $requestedName

weierophinney committed:
Minor updates to "$requestedName as SessionManager" feature

- Assign new variable name vs re-use of `$requestedName`.
- Use ternary for initial assignment of `$managerClass`.
- Space after `!` operator
- Updated test names to reflect behavior being tested.

weierophinney committed:
Added CHANGELOG for #73

weierophinney committed:
Merge branch 'feature/73' into develop

Close #73

weierophinney committed:
Merge pull request #78 from Slamdunk/develop_72

Add PHP 7.1 & 7.2 support, drop PHP 5.5 & HHVM, add PHPUnit 6 support

Conflicts:
	.travis.yml
	composer.json
	src/Config/SessionConfig.php
	src/Config/StandardConfig.php
	test/Config/SessionConfigTest.php
	test/Config/StandardConfigTest.php
	test/SessionManagerTest.php
	test/StorageTest.php

weierophinney committed:
Improved build matrix

Updated build matrix to follow lowest/locked/latest strategy, which will
make testing easire going forward.

weierophinney commented:
@Slamdunk I've updated this to remove conflicts, and to improve the build matrix.

weierophinney committed:
Updated new exception expectation to use `expectException()`

New expectation introduced in #73 to use modern exception expectation
method.

weierophinney commented:
Thanks, @Slamdunk; merged to develop for release with 2.8.0
